engine/pokemon/search.asm

```diff
@@ -1,81 +1,35 @@
 SpecialBeastsCheck:
-; Check if the player owns all three legendary beasts.
-; They must exist in either party or PC, and have the player's OT and ID.
+; Check if the player has caught all three legendary beasts.
 ; Return the result in hScriptVar.
 
-	ld a, RAIKOU
-	ldh [hScriptVar], a
-	call CheckOwnMonAnywhere
-	jr nc, SpecialMonCheckFailed
-
-	ld a, ENTEI
-	ldh [hScriptVar], a
-	call CheckOwnMonAnywhere
-	jr nc, SpecialMonCheckFailed
-
-	ld a, SUICUNE
-	ldh [hScriptVar], a
-	call CheckOwnMonAnywhere
-	jr nc, SpecialMonCheckFailed
-
-	; they exist
+	ld a, [wPlayerCaught]
+	and CAUGHT_BEASTS_MASK
+	cp CAUGHT_BEASTS_MASK
```

Comment:
https://github.com/pret/pokecrystal/wiki/Optimizing-assembly-code#compare-a-to-a-mask-after-masking-it